Src/PCompiler/PCommandLine/Options/PCompilerOptions.cs

```diff
@@ -103,6 +107,27 @@ private static void FindLocalPProject(List<CommandLineArgument> result)
             }
         }
 
+        /// <summary>
+        /// Updates the checkerConfiguration with the specified P project file.
+        /// </summary>
+        private static void UpdateConfigurationWithPProjectFile(CompilerConfiguration compilerConfiguration, List<CommandLineArgument> result)
+        {
+            foreach (var option in result)
+            {
+                switch (option.LongName)
+                {
+                    case "pproj":
+                    {
+                        new ParsePProjectFile().ParseProjectFile((string)option.Value, out var parsedConfig);
+                        compilerConfiguration.Copy(parsedConfig);
+                    }
+                        result.Remove(option);
```

Comment:
I dont understand this part? Why are you mutating the list you are iterating over?

Reply:
I am processing pproj argument first, and then other CLI arguments. Once pproj file is loaded, I was removing and returning the configuration loaded from pproj file, and next, to be configured/overridden by other CLI arguments.

I do see your point. Through c8f952f17bfc7484c5b7115c50fe7df305d65c25, I have cleaned this up so as to simply skip the already-loaded pproj argument (instead of mutating/removing) when parsing other CLI arguments.